TombEngine/Objects/TR2/Entity/tr2_spear_guardian.cpp

```diff
@@ -504,11 +505,19 @@ namespace TEN::Entities::Creatures::TR2
 						item->Animation.TargetState = SPEAR_GUARDIAN_STATE_SLASH_IDLE;
 				}
 				else if (ai.ahead && ai.distance < SPEAR_GUARDIAN_RUN_RANGE)
+				{
 					item->Animation.TargetState = SPEAR_GUARDIAN_STATE_WALK;
+				}
 				else
+				{
 					item->Animation.TargetState = SPEAR_GUARDIAN_STATE_RUN;
+				}
 
 				break;
+
+			case SPEAR_GUARDIAN_STATE_KILL:
+				creature->MaxTurn = 0;
```

Comment:
As above - why `MaxTurn` is zeroed here?

Reply:
Similar case than Giant Mutant, is set to 0 to stop the enemy for rotating while is doing the killing animation. Othewise can happen (what was happening!) that the spear guardian impaled Lara but then he faced to other direction, so Lara died floating in the air instead of being lifted by the spears. This fix prevent that from happening.